nucliadb_node/src/writer/grpc_driver.rs

```diff
@@ -269,16 +275,19 @@ impl NodeWriter for NodeWriterGRPCDriver {
         }
     }
 
-    async fn remove_resource(
+    async fn delete_relation_nodes(
         &self,
-        request: Request<ResourceId>,
+        request: Request<DeleteGraphNodes>,
     ) -> Result<Response<OpStatus>, Status> {
-        let resource = request.into_inner();
-        let shard_id = resource.shard_id.clone();
-        let shard = self.obtain_shard(&shard_id).await?;
+        let request = request.into_inner();
+        let shard_id = match request.shard_id {
+            Some(ref shard_id) => &shard_id.id,
+            None => return Err(tonic::Status::invalid_argument("Shard ID must be provided")),
```

Comment:
I think I would have gone for:
```rust
        let Some(shard_id) = request.shard_id.as_ref().map(|i| &i.id) else {
            return Err(tonic::Status::invalid_argument("Shard ID must be provided"));
        };
```

Reply:
This is code moved inside the file and the style is the same across all functions. I don't see much benefit changing it